File: Python/problem14.py
import re

str = 'mouseratmouse'

def solution(s):
    mouse_pattern = re.compile('mouse', re.IGNORECASE)
    rat_pattern = re.compile('rat', re.IGNORECASE)
    
    mouse_matches = mouse_pattern.findall(s)
    rat_matches = rat_pattern.findall(s)
    
    print(mouse_matches)
    print(rat_matches)
    
    print(len(mouse_matches) == len(rat_matches))


solution(str)

File: JavaScript/problem14.js
const str = 'mouseratmouse'

function solution(str) {
    mouseCount = str.match(/mouse/gi)
    console.log(mouseCount)
    ratCount = str.match(/rat/gi)
    console.log(ratCount)

    console.log(mouseCount.length == ratCount.length)
}

solution(str)